### DISCUSSION TITLE
How to combine `InstrumentedAttribute[T].any`  with a `Select`

### DISCUSSION THREAD

**Discussion by nrbnlulu** [sqlalchemy/sqlalchemy#11441] [Usage Questions] [answered]
I have a usecase where I need to combine a statement with an `InstrumentedAttribute`

Something like:
```py
stmt = stmt.where(
	SomeTable.O2M_field.any(
			another_statement
	),
)
```
Though it seems that `InstrumentedAttribute.any` accepts only column expressions.
Is there a way around it?

**Comment by nrbnlulu** [comment, marked as answer]
I think I'v found it
```py
stmt = stmt.where(
	SomeTable.O2M_field.any(
			another_statement.whereclause
	),
)
```
pyright is happy now. I'll update if it works as well :upside_down_face: 

So generally it workes if you use the orm. But if you want to query something inside another query you'd use what @CaselIT suggested
[here](https://github.com/sqlalchemy/sqlalchemy/discussions/11441#discussioncomment-9636416)

**Comment by CaselIT** [comment]
Hi,

It's not clear what `another_statement` is

**Reply by nrbnlulu** [reply]
`sa.Select`

**Reply by CaselIT** [reply]
The typing is correct. You need to call `scalar_subquery()` to use a select in a column context

**Reply by nrbnlulu** [reply]
hmm I don't wan't the inner select just the clause... the orm handles this I think
